test/unit/services/test_manufacturer.py

```diff
@@ -2,237 +2,382 @@
+    """
+    Base class for create tests.
+    """
+
+    _manufacturer_post: ManufacturerPostSchema
+    _expected_manufacturer_in: ManufacturerIn
+    _expected_manufacturer_out: ManufacturerOut
+    _created_manufacturer: ManufacturerOut
+    _create_exception: pytest.ExceptionInfo
+
+    def mock_create(self, manufacturer_post_data: dict) -> None:
+        """
+        Mocks repo methods appropriately to test the `create` service method.
+
+        :param manufacturer_post_data: Dictionary containing the basic manufacturer data as would be required for a
+            `ManufacturerPostSchema` (i.e. no ID, code or created and modified times required).
+        """
+        self._manufacturer_post = ManufacturerPostSchema(**manufacturer_post_data)
+
+        self._expected_manufacturer_in = ManufacturerIn(
+            **manufacturer_post_data, code=utils.generate_code(manufacturer_post_data["name"], "manufacturer")
+        )
+        self._expected_manufacturer_out = ManufacturerOut(**self._expected_manufacturer_in.model_dump(), id=ObjectId())
+
+        ServiceTestHelpers.mock_create(self.mock_manufacturer_repository, self._expected_manufacturer_out)
+
+    def call_create(self) -> None:
+        """
+        Calls the `ManufacturerService` `create` method with the appropriate data from a prior call to `mock_create`.
+        """
+        self._created_manufacturer = self.manufacturer_service.create(self._manufacturer_post)
+
+    def check_create_success(self) -> None:
+        """
+        Checks that a prior call to `call_create` worked as expected.
+        """
+        self.wrapped_utils.generate_code.assert_called_once_with(self._expected_manufacturer_out.name, "manufacturer")
+        self.mock_manufacturer_repository.create.assert_called_once_with(self._expected_manufacturer_in)
+        assert self._created_manufacturer == self._expected_manufacturer_out
+
+
+class TestCreate(CreateDSL):
+    """
+    Tests for creating a manufacturer.
+    """
+
+    def test_create(self):
+        """
+        Test creating a manufacturer.
+        """
+        self.mock_create(MANUFACTURER_POST_DATA_A)
+        self.call_create()
+        self.check_create_success()
+
+
+class GetDSL(ManufacturerServiceDSL):
+    """
+    Base class for get tests.
+    """
+
+    _obtained_manufacturer_id: str
+    _expected_manufacturer: MagicMock
+    _obtained_manufacturer: MagicMock
+
+    def mock_get(self) -> None:
+        """
+        Mocks repo methods appropriately to test the `get` service method.
+        """
+        # Simply a return currently, so no need to use actual data
+        self._expected_manufacturer = MagicMock()
+        ServiceTestHelpers.mock_get(self.mock_manufacturer_repository, self._expected_manufacturer)
+
+    def call_get(self, manufacturer_id: str) -> None:
+        """
+        Calls the `ManufacturerService` `get` method.
+
+        :param manufacturer_id: ID of the manufacturer that will be obtained.
```

Comment:
```suggestion
        :param manufacturer_id: ID of the manufacturer to be obtained.
```
Change to fit with update (and my own change while adding these comments everywhere else).